### PR TITLE
Bug 1249828 Correct title for synced tabs

### PR DESCRIPTION
**Pull request by fluffyemily**
There are 2 issues in this bug.

1) Tabs are being synced with the URL as the title.

The cause of this is that when we restore tabs, as soon as the webview is created we delete the lastTitle value. However, if we subsequently don't load a URL for that restored tab, then the lastTitle is nil. In the case that the tab we are saving for sync doesn't have a title (i.e, has never been loaded in this browsing session) then when we come to save, we prioritize the URL over the lastTab, which in any case is empty.

To resolve this, I have ensured that we don't nullify the lastTitle variable when creating a webView and prioritized lastTitle over the URL in the case of the webView having no title.

2) Tabs are being synced with localhost URLs

The cause of why we are attempting to store localhost URLs is uncertain - I have been unable to replicate this case. To cover our back though, I have added some code here that fails to create a `RemoteTab` instance on sync tab save if the main URL is in the ignore list.

### CODE REVIEW

File: Client/Frontend/Browser/Browser.swift [hunk 1]
```diff
@@ -80,7 +80,7 @@ class Browser: NSObject, BrowserWebViewDelegate {
     }
 
     class func toTab(browser: Browser) -> RemoteTab? {
-        if let displayURL = browser.displayURL {
+        if let displayURL = browser.displayURL where !isIgnoredURL(displayURL) {
```
Comment (rnewman) [inline Client/Frontend/Browser/Browser.swift]: `where RemoteTab.shouldIncludeURL(displayURL)`

and below.

File: Client/Frontend/Browser/Browser.swift [hunk 2]
```diff
@@ -208,7 +206,12 @@ class Browser: NSObject, BrowserWebViewDelegate {
                 return title
             }
         }
-        return displayURL?.absoluteString ?? lastTitle ?? ""
+
+        guard let lastTitle = lastTitle where !lastTitle.isEmpty else {
+            return  displayURL?.absoluteString ??  ""
```
Comment (rnewman) [inline Client/Frontend/Browser/Browser.swift]: Nit: remove double space.

File: Client/Frontend/Browser/Browser.swift (hunk 1 above)
Comment (fluffyemily) [inline Client/Frontend/Browser/Browser.swift]: woah. I totally didn't see that there! awesome :)

File: Client/Frontend/Browser/Browser.swift [hunk 3]
```diff
@@ -90,12 +90,14 @@ class Browser: NSObject, BrowserWebViewDelegate {
                 icon: nil)
         } else if let sessionData = browser.sessionData where !sessionData.urls.isEmpty {
             let history = Array(sessionData.urls.reverse())
```
Comment (rnewman) [inline Client/Frontend/Browser/Browser.swift]: This should have the same `filter` call as above.

**Comment by rnewman** [comment]
I think not nulling the title is safe, assuming we'll always get a new title after the view loads the right page, yes?

File: Client/Frontend/Browser/Browser.swift (hunk 1 above)
Comment (rnewman) [inline Client/Frontend/Browser/Browser.swift]: They're very similar, in fairness!

**Comment by fluffyemily** [comment]
yeah, we will always return the title of the webView if present, so if a new title is loaded, this becomes our new title when we save our tabs, both for restoration and syncing purposes.

**Comment by rnewman** [comment]
LGTM with those nits!

**Pull request merged**